enhancements/proposals/2490-scylladbmonitoring-external-prometheus/README.md

```diff
@@ -0,0 +1,611 @@
+       // +optional
+       PrometheusOptions *GrafanaPrometheusDatasourceOptions `json:"prometheusOptions,omitempty"`
+   }
+   ```
+
+   Strongly-typed enum will be used for the datasource type:
+
+   ```go
+   // GrafanaDatasourceType defines the type of Grafana datasource.
+   // +kubebuilder:validation:Enum="Prometheus"
+   type GrafanaDatasourceType string
+
+   const (
+       // GrafanaDatasourceTypePrometheus is the Prometheus datasource type.
+       GrafanaDatasourceTypePrometheus GrafanaDatasourceType = "Prometheus"
+   )
+   ```
+
+   `GrafanaPrometheusDatasourceSpec` will define Prometheus-specific options, including TLS and authentication.
+
+   ```go
+   type GrafanaPrometheusDatasourceOptions struct {
+      // tls holds TLS configuration for connecting to Prometheus over HTTPS.
+      // +optional
+      TLS *GrafanaDatasourceTLSSpec `json:"tls,omitempty"`
+
+      // auth holds authentication options for connecting to Prometheus.
+      // +optional
+      Auth *GrafanaPrometheusDatasourceAuthSpec `json:"auth,omitempty"`
+   }
+   ```
+
+   `GrafanaDatasourceTLSSpec` will define TLS configuration options for server certificate verification and client
+   authentication.
+
+   ```go
+   // LocalObjectKeySelector selects a key of a ConfigMap or Secret in the same namespace.
+   type LocalObjectKeySelector struct {
+      // name of the selected object.
+      // +kubebuilder:validation:MinLength=1
+      Name string `json:"name"`
+
+      // key within the selected object.
+      // +kubebuilder:validation:MinLength=1
+      Key string `json:"key"`
+   }
+
+   type GrafanaDatasourceTLSSpec struct {
+      // caCert is a reference to a key within the CA bundle ConfigMap. The key should hold the CA cert in PEM format.
+      // When not specified, system CAs are used.
+      // +optional
+      CACertConfigMapRef *LocalObjectKeySelector `json:"caCertConfigMapRef,omitempty"`
+
+      // insecureSkipVerify controls whether to skip server certificate verification.
+      // +kubebuilder:default:=false
+      // +optional
+      InsecureSkipVerify bool `json:"insecureSkipVerify,omitempty"`
+
+	  // clientTLSKeyPairSecretRef is a name of a Secret holding client TLS certificate and key for mTLS authentication.
+	  // It's expected to be a standard Kubernetes TLS Secret with `tls.crt` and `tls.key` keys.
+      // +optional
+      ClientTLSKeyPairSecretName *corev1.LocalObjectReference `json:"clientTLSKeyPairSecretRef,omitempty"`
+   }
+   ```
+
+   `GrafanaPrometheusDatasourceAuthSpec` will define authentication options, starting with bearer token auth. It will
+   be a separate struct to allow adding more auth methods in the future if needed (e.g., basic auth).
+
+   ```go
+   type GrafanaPrometheusDatasourceAuthType string
+   
+   const (
+      // GrafanaPrometheusDatasourceAuthTypeNone means no authentication.
+      GrafanaPrometheusDatasourceAuthTypeNone GrafanaPrometheusDatasourceAuthType = "None"
```

Comment:
How about we call it Unauthorized or NoAuthentication to better reflect what is being set?